lib/appsignal_phoenix/event_handler.ex

```diff
@@ -45,11 +45,8 @@ defmodule Appsignal.Phoenix.EventHandler do
     |> @span.set_attribute("appsignal:category", "call.phoenix_endpoint")
   end
 
-  def phoenix_endpoint_stop(_event, _measurements, metadata, _config) do
-    _root_span = set_span_data(@tracer.root_span(), metadata)
-
-    @tracer.current_span()
-    |> @tracer.close_span()
+  def phoenix_endpoint_stop(_event, _measurements, _metadata, _config) do
+    @tracer.close_span(@tracer.current_span())
```

Comment:
If this event doesn't get called, does that mean the span doesn't get closed either? Can that be the root span that doesn't get closed?

Reply:
That means the endpoint span doesn't get closed, but that also doesn't exist because the endpoint start event also doesn't fire. The problem is not that the root span doesn't get closed, it's that the span metadata isn't set.

Reply:
Ok makes sense 👍